utils/src/linked_list.c

```diff
@@ -0,0 +1,230 @@
+#include "linked_list.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+/**
+ * @brief      Prints data of a linked list.
+ *
+ * @param[in]      list  The list to print
+ */
+void print_linked_list(linked_list_t *list) {
+    // Check for NULL
+    if (list == NULL) {
+        return;
+    }
+    node_t *curr_node = NULL;
+    for (curr_node = list->head; curr_node != NULL;
+         curr_node = curr_node->next) {
+        printf("%s\n", (char *)curr_node->data);
+    }
+}
+
+/**
+ * @brief      Initializes the linked list
+ *
+ * @return     Returns an empty linked list
+ */
+linked_list_t *init_list() {
+    linked_list_t *list = (linked_list_t *)calloc(1, sizeof(linked_list_t));
+
+    // Check NULL
+    if (list == NULL) {
+        return NULL;
+    }
+    return list;
+}
+
+/**
+ * @brief      Returns the data from the tail of the list
+ *
+ * @param[in]      list  The list
+ *
+ * @return     Returns the last item in the list
+ */
+void *get_tail(linked_list_t *list) {
+    // Check for NULL
+    if (list == NULL) {
+        return NULL;
+    }
+
+    // Return tail data if not NULL
+    if (list->tail != NULL) {
+        return list->tail->data;
+    }
+
+    return NULL;
+}
+
+/**
+ * @brief      Insert data into the given linked list
+ *
+ * @param[in]  list  The list for insertion
+ * @param[in]  data  The data to insert
+ *
+ * @return     Returns the node that the data was inserted into else NULL if not
+ * found
+ */
+node_t *insert_data(linked_list_t *list, void *data) {
+    // Check for NULL
+    if (list == NULL) {
+        return NULL;
+    }
+
+    node_t *new_node = (node_t *)calloc(1, sizeof(node_t));
```

Comment:
You shouldn't need to ``calloc`` if all the fields get assigned too.

Reply:
You also dont need to cast the result of ``calloc`` to ``node_t*``.  However, if this is for readability it's fine with me.